File: mycollab-ui/src/main/scala/com/mycollab/vaadin/mvp/ScreenData.scala
package com.mycollab.vaadin.mvp

import com.mycollab.db.arguments.SearchCriteria

import scala.beans.BeanProperty

/**
  * @author MyCollab Ltd.
  * @since 5.0.3
  */
class ScreenData[P](@BeanProperty var params: P) {}

object ScreenData {

  class Add[P](params: P) extends ScreenData[P](params) {}

  class Edit[P](params: P) extends ScreenData[P](params) {}

  class Preview[P](params: P) extends ScreenData[P](params) {}

  class Search[P <: SearchCriteria](params: P) extends ScreenData(params) {}

}

File: mycollab-ui/src/main/java/com/mycollab/vaadin/mvp/ScreenData.kt
package com.mycollab.vaadin.mvp

import com.mycollab.db.arguments.SearchCriteria

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
open class ScreenData<P>(var params: P?) {

    class Add<P>(params: P?) : ScreenData<P>(params)

    class Edit<P>(params: P?) : ScreenData<P>(params)

    class Preview<P>(params: P?) : ScreenData<P>(params)

    class Search<P : SearchCriteria>(params: P?) : ScreenData<P>(params)
} 